spec/teacher_spec.rb

```diff
@@ -0,0 +1,10 @@
+require_relative '../teacher'
+
+describe Teacher do
+  let(:teacher) { described_class.new('Math', 30, parent_permission: true, name: 'John Doe') }
+
+  it 'checks attributes' do
+    expect(teacher.age).to eq(0)
+    expect(teacher.name).to eq('John Doe')
+  end
```

Comment:
- [ ] Also, for the Teacher class we need to test the `can_use_services`, This is to assert our test covers all the methods in all the classes.